### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by oriubimo**

### CODE REVIEW

File: mr/readme.md [hunk 1]
```diff
@@ -110,8 +112,8 @@ Some forms of IO storage (for example, Google Storage) work better when you read
 
 Note that the idealized model of a thread per CPU doesn't actually work on Linux when reading files from local disk. Linux doesn't support async IO for any filesystem that is not XFS. Because of this, there is a separate pool of threads that only run IO calls and send their results to the caller. This problem doesn't exist when working with Google Storage, since Asio is capable of handling asynchronous network IO.
 
-TODO: Write about runner.
+Although both types of executors handle the logic of the mapping/joining process, they try to avoid doing the I/O operation directly. Instead, I/O operations are implemented by a `Runner` object which represents the interactions of the MR infrastructure with the disk/net.
```
Comment (romange) [inline mr/readme.md]: they avoid interacting with the external environment directly. Instead, all I/O operations  are abstracted away using a `Runner` object which represents the external environment of the MR infrastructure.

File: mr/readme.md [hunk 2]
```diff
@@ -110,8 +112,8 @@ Some forms of IO storage (for example, Google Storage) work better when you read
...
 
-TODO: Write about HandlerWrapper.
\ No newline at end of file
+In the previous paragraph, `RawContext` objects were mentioned. `RawContext` can be thought of as a part of `Runner` that is given to worker threads and allows them to write outputs from their MR execution into files (in fact, the `DoContext` that user-written functions accept is a simple wrapper around `RawContext`). Note, however, that `RawContext`, unlike `Runner`, is not purely an abstract interface, it also contains logic to handle freq counters instead of only supporting I/O operations.
```
Comment (romange) [inline mr/readme.md]: ....`RawContext` objects were mentioned. `RawContext` is the channel connecting pipeline user code (operator implementations) to the framework.  `RawContext` is handled by `Runner` and among other things it allows operators to write their outputs into files (in fact ...

**Comment by oriubimo** [comment]
Ich habe die aufforderte Lösungen angewendet

**Comment by oriubimo** [comment]
Applied the requested fixes.

**Pull request merged**